src/admin/bundles/bundle-install.directive.js

```diff
@@ -0,0 +1,124 @@
+(function(){
+    'use-strict';
+
+    angular.module('motech-admin')
+        .directive('installModules', moduleInstall);
+
+    moduleInstall.$inject = ['ServerService'];
+    function moduleInstall(ServerService) {
+        return {
+            restrict: 'E',
+            replace: true,
+            templateUrl: '/admin/bundles/install-modules.html',
+            controller: controller,
+            link: function(scope, element, attrs) {
+                var url = $("#bundleUploadForm").attr("action");
+                url = ServerService.formatURL(url);
+                $("#bundleUploadForm").attr("action", url);
+            }
+        };
+    }
+
+    controller.$inject = ['$scope', '$rootScope', '$state', 'BundlesFactory', 'LoadingModal', 'ModalFactory'];
+    function controller($scope, $rootScope, $state, BundlesFactory, LoadingModal, ModalFactory) {
+        $scope.moduleSources = [
+            'Repository',
+            'File'
+        ];
+
+        $scope.moduleSource = $scope.moduleSources[0];
+
+        $scope.mavenStr = function(artifactId) {
+            return 'org.motechproject:'.concat(artifactId).concat(':').concat($scope.msg('server.version'));
+        };
+
+        $scope.modules = {};
+        $scope.modules[$scope.mavenStr('alerts')] = 'Alerts';
+        $scope.modules[$scope.mavenStr('appointments')] = 'Appointments';
+        $scope.modules[$scope.mavenStr('atom-client')] = 'Atom Client';
+        $scope.modules[$scope.mavenStr('csd')] = 'Care Services Directory';
+        $scope.modules[$scope.mavenStr('cms-lite')] = 'CMS Lite';
+        $scope.modules[$scope.mavenStr('commcare')] = 'Commcare';
+        $scope.modules[$scope.mavenStr('dhis2')] = 'DHIS2';
+        $scope.modules[$scope.mavenStr('event-logging')] = 'Event Logging';
+        $scope.modules[$scope.mavenStr('http-agent')] = 'Http Agent';
+        $scope.modules[$scope.mavenStr('ihe-interop')] = 'IHE Interop';
+        $scope.modules[$scope.mavenStr('ivr')] = 'IVR';
+        $scope.modules[$scope.mavenStr('message-campaign')] = 'Message Campaign';
+        $scope.modules[$scope.mavenStr('metrics')] = 'Metrics';
+        $scope.modules[$scope.mavenStr('mtraining')] = 'mTraining';
+        $scope.modules[$scope.mavenStr('motech-tasks')] = 'Tasks';
+        $scope.modules[$scope.mavenStr('odk')] = 'Open Data Kit';
+        $scope.modules[$scope.mavenStr('openmrs')] = 'OpenMRS';
+        $scope.modules[$scope.mavenStr('pill-reminder')] = 'Pill Reminder';
+        $scope.modules[$scope.mavenStr('motech-scheduler')] = 'Scheduler';
+        $scope.modules[$scope.mavenStr('schedule-tracking')] = 'Schedule Tracking';
+        $scope.modules[$scope.mavenStr('sms')] = 'SMS';
+
+        $scope.module = "";
+
+        $scope.startOnUpload = function () {
+            if ($scope.startUpload !== true) {
+                $scope.startUpload = true;
+                $('.start-on-upload').find('i').removeClass("fa-square-o").addClass('fa-check-square-o');
```

Comment:
Can we remove the CSS manipulation from the controller and either do it in the markup or a directive link function?